lib/rucio/tests/test_replica.py

```diff
@@ -83,6 +85,57 @@ def setUp(self):
         else:
             self.vo = {}
 
+    def test_cache_replicas(self):
+        """ REPLICA (CORE): Test listing replicas with cached root protocol """
+
+        rse = 'APERTURE_%s' % rse_name_generator()
+        rse_id = add_rse(rse, **self.vo)
+
+        add_protocol(rse_id, {'scheme': 'root',
+                              'hostname': 'root.aperture.com',
+                              'port': 1409,
+                              'prefix': '//test/chamber/',
+                              'impl': 'rucio.rse.protocols.xrootd.Default',
+                              'domains': {
+                                  'lan': {'read': 1, 'write': 1, 'delete': 1},
+                                  'wan': {'read': 1, 'write': 1, 'delete': 1}}})
+        add_protocol(rse_id, {'scheme': 'http',
+                              'hostname': 'root.aperture.com',
+                              'port': 1409,
+                              'prefix': '//test/chamber/',
+                              'impl': 'rucio.rse.protocols.xrootd.Default',
+                              'domains': {
+                                  'lan': {'read': 1, 'write': 1, 'delete': 1},
+                                  'wan': {'read': 1, 'write': 1, 'delete': 1}}})
+
+        tmp_scope = InternalScope('mock', **self.vo)
+        root = InternalAccount('root', **self.vo)
+
+        files = []
+
+        name = 'file_%s' % generate_uuid()
+        hstr = hashlib.md5(('%s:%s' % (tmp_scope, name)).encode('utf-8')).hexdigest()
+        pfn = 'root://root.aperture.com:1409//test/chamber/mock/%s/%s/%s' % (hstr[0:2], hstr[2:4], name)
+        files.append({'scope': tmp_scope, 'name': name, 'bytes': 1234, 'adler32': 'deadbeef', 'pfn': pfn})
+
+        name = 'element_%s' % generate_uuid()
+        hstr = hashlib.md5(('%s:%s' % (tmp_scope, name)).encode('utf-8')).hexdigest()
+        pfn = 'http://root.aperture.com:1409//test/chamber/mock/%s/%s/%s' % (hstr[0:2], hstr[2:4], name)
+        files.append({'scope': tmp_scope, 'name': name, 'bytes': 1234, 'adler32': 'deadbeef', 'pfn': pfn})
+
+        add_replicas(rse_id=rse_id, files=files, account=root)
+
+        cconfig_set('clientcachemap', 'BLACKMESA', 'AGLT2')
+        cconfig_set('virtual_placement', 'vp_endpoint', 'https://vps.cern.ch')
```

Comment:
again, mock this for the tests. This may or may not fail when the server is not available. You could mock/patch the requests.get method for testing `get_multi_cache_prefix`. It would also be nice to have tests for error cases.

Reply:
Yes, I agree. Although this is a prototype feature for now, it would be better to mock this right away. The vps server might be unavailable, or it might not be reachable from the GH test farm etc. and then our tests fail because of it. Better to mock it right away.

Reply:
this will fail if the server is not available. it must return a prefix which is not '', that is what is looked for in the assert.